src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -411,6 +416,10 @@ public void onDeath(@NotNull final DamageSource damageSource)
             citizenColonyHandler.getColony().decreaseOverallHappiness(penalty);
             triggerDeathAchievement(damageSource, citizenJobHandler.getColonyJob());
             citizenChatHandler.notifyDeath(damageSource);
+            if (!(citizenJobHandler.getColonyJob() instanceof AbstractJobGuard))
+            {
+                citizenColonyHandler.getColony().setNeedToMourn(true, citizenData.getName());
+            }
             citizenColonyHandler.getColony().getCitizenManager().removeCitizen(getCitizenData());
```

Comment:
I think it would be better to just let them mourn when the player killed the entity, else people could be quite angry about how often their citizens don't work.

Reply:
@Raycoms That could be changed.   I was thinking more of needing guards to protect your colony and making citizens happy.  But if citizens are dying that often then it could need up making them mourn everyday.

Got to ask,  in my colony I might have a guard die here and their but a regular citizen hasn't died in the last 30 days on my game.    But I see people are killing there citizens all the time.  Didn't get that either.  I guess people want faster leveling up and just keeps routing citizens until they get a good skill level.

Also, see all the time "I fired him" because he was not working.  That would mean bug, or them not understanding.

Reply:
Can we remove the inWall deaths here again, we still are having them, else I think it looks good.
How long do they mourn?

Reply:
They mourn for 2,000 - 4000 ticks..  I reduced it down from all day to that.   It just a partial day and not all of them mourn for the same amount.

I'll have to check to see where we have to do that check and how we pass it around so the ondeath routine will not call mourn on those deaths, but it should be easy.